### PR TITLE
[V3 Filter] Throw a filter Event on_filter_message_delete

### PR DESCRIPTION
**Pull request by entchen66**
reference to https://github.com/Cog-Creators/Red-DiscordBot/pull/2443

### Type

- [ ] Bugfix
- [ ] Enhancement
- [x] New feature

### Description of the changes

Like Sinbad and Kowlin talked in PR 2443, I now add an event to the filter cog. It is thrown when a message got deleted and provide the message to custom cogs, just listen to the on_filter_message_delete event.

### CODE REVIEW

File: redbot/cogs/filter/filter.py [hunk 1]
```diff
@@ -363,6 +363,7 @@ def __unload(self):
 
         if hits:
             try:
+                self.bot.dispatch("filter_message_delete", message)
```
Comment (Tobotimus) [inline redbot/cogs/filter/filter.py]: The event should only be dispatched if the deletion succeeds. To do this, move this line down to 371, just after the `else:` line :smiley:

**Review by Tobotimus — CHANGES_REQUESTED** [review]
Thanks for taking the time to open this PR. Just would like you to change it a bit :smile:

**Comment by Tobotimus** [comment]
It might also be useful to provide `hits` as an argument to event listeners, so for example with your logging use case, the log message could display exactly which words/phrases were detected in the message.

File: redbot/cogs/filter/filter.py (hunk 1 above)
Comment (entchen66) [reply redbot/cogs/filter/filter.py]: oh yeah, sure^^

File: redbot/cogs/filter/filter.py [hunk 2]
```diff
@@ -364,6 +364,7 @@ def __unload(self):
         if hits:
             try:
                 await message.delete()
+                self.bot.dispatch("filter_message_delete", message, hits)
```
Comment (Tobotimus) [inline redbot/cogs/filter/filter.py]: Please move it to line 371, after the `else:` line :D

**Review by Tobotimus — CHANGES_REQUESTED** [review]

**Review by Tobotimus — APPROVED** [review]
Thanks for your contribution @entchen66 :smiley:

**Pull request merged**